lib/models/LocationStatus.ts

```diff
@@ -0,0 +1,143 @@
+ */
+export default class LocationStatus {
+    _data: LocationServiceStatus;
+
+    constructor(locationStatus?: Object | LocationServiceStatus) {
+        this._data = this._initStatus();
+        if (locationStatus) {
+            let data;
+            if (locationStatus instanceof LocationStatus) {
+                data = locationStatus._data;
+            } else {
+                data = locationStatus;
+            }
+            Object.assign(this._data, JSON.parse(JSON.stringify(data)));
+        }
+    }
+
+    /**
+     * Initializes the status of all services
+     * The default status of a service is unpaused
+     * @returns {LocationServiceStatus}
+     */
+    _initStatus(): LocationServiceStatus {
+        return {
+            replication: {
+                paused: false,
+                scheduledResume: null,
+            },
+            ingestion: {
+                paused: false,
+                scheduledResume: null,
+            },
+            lifecycle: {
+                paused: false,
+                scheduledResume: null,
+            },
+        };
+    }
+
+    /**
+     * @param service service name
+     * @param paused true if paused
+     */
+    setServicePauseStatus(service: string, paused: boolean) {
+        if (Object.keys(this._data).includes(service)) {
+            this._data[service].paused = paused;
+        }
+    }
+
+    /**
+     * @param service service name
+     * @returns true if paused
+     */
+    getServicePauseStatus(service: string) : boolean | null {
+        if (!Object.keys(this._data).includes(service)) {
+            return null;
+        }
+        return this._data[service].paused;
+    }
+
+    /**
+     * @param service service name
+     */
+    setServiceResumeSchedule(service: string, date: Date | null) {
+        if (Object.keys(this._data).includes(service)) {
+            if (date !== null) {
+                this._data[service].scheduledResume = date.toString();
+            } else {
+                this._data[service].scheduledResume = null;
+            }
+        }
+    }
+
+    /**
+     * @param service service name
+     * @returns scheduled resume date
+     */
+    getServiceResumeSchedule(service: string) : Date | null {
+        const schedule = this._data[service].scheduledResume;
+        if (!schedule) {
+            return null;
+        }
+        return new Date(schedule);
+    }
+
+    /**
+     * @param service service(s) name
+     */
+    pauseLocation(service: string | string[]) {
+        const servicesList = Array.isArray(service) ?
+            service : [service];
+        servicesList.forEach(svc => {
+            this.setServicePauseStatus(svc, true);
+        });
+    }
+
+    /**
+     * @param service service(s) name
+     * @param schedule date to resume service(s)
+     */
+    resumeLocation(service: string | string[], schedule?: Date) {
+        const servicesList = Array.isArray(service) ?
```

Comment:
can we resume a location if the location is not paused? I am thinking if we need to first check if a location is paused, then resume it

Reply:
I see now, you checked [here](https://github.com/scality/backbeat/pull/2361/commits/d54b244609ff97e235483c7c9035c3edf0ba8e37#diff-feb400da3fd9b9f1e83c06416f27265ba93371e0f311b75512ec575ebd590b51R1267)